lib/Zonemaster/Backend/DB/PostgreSQL.pm

```diff
@@ -143,17 +141,10 @@ sub create_new_test {
 
     my $nb_inserted = $dbh->do( $query );
 
-    my ( $id, $hash_id ) = $dbh->selectrow_array(
-        "SELECT id, hash_id FROM test_results WHERE params_deterministic_hash='$test_params_deterministic_hash' ORDER BY id DESC LIMIT 1" );
-        
-    if ( $id > $self->config->force_hash_id_use_in_API_starting_from_id() ) {
-        $result = $hash_id;
-    }
-    else {
-        $result = $id;
-    }
+    my ( undef, $hash_id ) = $dbh->selectrow_array(
+        "SELECT id,hash_id FROM test_results WHERE params_deterministic_hash='$test_params_deterministic_hash' ORDER BY id DESC LIMIT 1" );
```

Comment:
Have you considered updating this to use `?` instead of string interpolation?

Reply:
Not really because I wanted to limit the change to what is needed. Would that be a considerable improvement? If so, I could do that change.

Reply:
Yes, it would definitely be an improvement. It makes it much easier to make code reviews with regard to SQL injections.

Reply:
In the updated commits the change is already implemented.